opennebula/resource_opennebula_template.go

```diff
@@ -424,9 +424,49 @@ func resourceOpennebulaTemplateReadCustom(ctx context.Context, d *schema.Resourc
 	d.Set("gid", tpl.GID)
 	d.Set("uname", tpl.UName)
 	d.Set("gname", tpl.GName)
+	d.Set("group", tpl.GName)
 	d.Set("reg_time", tpl.RegTime)
+	d.Set("description", "")
+	d.Set("sched_requirements", "")
 	d.Set("permissions", permissionsUnixString(*tpl.Permissions))
 
+	getTags := make(map[string]interface{}, 0)
+	for k, v := range pairsToMap(tpl.Template.Template) {
+		if k != "MEMORY" && k != "CPU" && k != "VCPU" && k != "VROUTER" && k != "DESCRIPTION" && k != "SCHED_REQUIREMENTS" {
+			getTags[strings.ToLower(k)] = v
+		}
+	}
+	d.Set("tags", getTags)
```

Comment:
In the method `flattenVMUserTemplate` there was a line `if tagsInterface, ok := d.GetOk("tags"); ok {`. Because of this, only tags which already were present in the state will be processed. I will add a commit where the method `flattenVMUserTemplate` is updated: it will have a loop over all elements in the template, instead of over all elements in the state.